pvm.py

```diff
@@ -35,116 +43,230 @@ def _init_logger():
 		if not IS_FILE_SET:
 			_logger.info("Command %s failed because of the file is unset.", command)
 			return
 
-		if command=="start":
-			if media.can_play():
-				media.play()
+		# If Raspberry Pi has two displays.
+		if displayNum == 2:
+			if command=="start":
+				if canStart:
+						# Get position for media1 and media2
+						pos1 = media1.position()
+						pos2 = media2.position()
+						# Always set position for the fast media to slow one.
+						if pos1 > pos2:
+							media1.set_position(pos2)
+						elif pos1 < pos2:
+							media2.set_position(pos1)		
+						pos1 = media1.position()
+						_logger.info("media 1&2 position: %s", pos1)
+						# Play both media
+						media1.play()
+						media2.play()
+						# Change canPause and canStart
+						canPause = True
+						canStart = False
+						_logger.info("%s command success.", command)
+				else:
+					_logger.info("%s command failed.", command)
+			elif command=="stop":
+				if media1.can_quit():
+					media1.quit()
+					media2.quit()
+					IS_FILE_SET = False
+					canPause = False
+					canStart = False
+					_logger.info("%s command success and file has been unset.", command)
+				else:
+					_logger.info("%s command failed.", command)
+			elif command=="set_position":
+				p1, p2 = parse_value(value)
+				media1.set_position(float(p1))
+				media2.set_position(float(p2))
 				_logger.info("%s command success.", command)
-			else:
-				_logger.info("%s command failed.", command)
-		elif command=="stop":
-			if media.can_quit():
-				media.stop()
-				IS_FILE_SET = False
-				_logger.info("%s command success and file has been unset.", command)
-			else:
-				_logger.info("%s command failed.", command)
-		elif command=="set_position":
-			media.set_position(float(value))
-			_logger.info("%s command success.", command)
-		elif command=="set_rate":
-			fps = str(30 * float(value))
-			media = OMXPlayer(PREFIX_PATH + VIDEO_PATH, dbus_name='org.mpris.MediaPlayer2.omxplayer', args=['--loop','--force-fps', fps])
-			if media != "":
-				media.quit()
-				print("media quit")
-			original_fps = 30
-			video_info = subprocess.Popen(["omxplayer", "-i", PEFIX_PATH + VIDEO_PATH], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-			print("The command", ["omxplayer", "-i", PEFIX_PATH + VIDEO_PATH])
-			# info = video_info.stdout.decode().split(", ")
-			out, err = video_info.communicate()
-			out = out.decode(encoding='utf-8')
-			splist = out.split(", ")
-			for s in splist:
-				if 'fps' in s:
-					print("current ele:", s)
-					fps_info = s.split(' ')
-					original_fps = float(fps_info[0])
-			fps = str(original_fps * float(value))
-			print("computed fps", fps)
-			media = OMXPlayer(PREFIX_PATH + VIDEO_PATH, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=['--loop','--force-fps', fps])
-			media.pause()
-			_logger.info("%s command success.", command)
-		elif command=="pause":
-			if media.can_pause():
-				media.pause()
+			elif command=="set_rate":
+				v1, v2 = parse_value(value)
```

Comment:
we are parsing values in lines 120, 176, and 181. Isn't it possible to just parse it once and be done with it? what is the advantage of doing it this way, all these variables will not stay alive for when the next command is parsed, right?